legacy/core/src/test/java/com/fsck/k9/helper/FileHelperTest.kt

```diff
@@ -0,0 +1,50 @@
+package com.fsck.k9.helper
+
+import assertk.assertThat
+import assertk.assertions.isFalse
+import assertk.assertions.isNotEqualTo
+import assertk.assertions.isTrue
+import java.io.File
+import kotlin.test.Test
+import org.junit.Rule
+import org.junit.rules.TemporaryFolder
+
+class FileHelperTest {
+
+    @Rule
+    @JvmField
+    val tempFolder = TemporaryFolder()
+
+    @Test
+    fun `touchFile should create file`() {
+        val parent = tempFolder.newFolder("parent")
+
+        FileHelper.touchFile(parent, "FileName")
+
+        assertThat(File(parent, "FileName").exists()).isTrue()
+    }
+
+    @Test
+    fun `touchFile should change file modification time`() {
+        val parent = tempFolder.newFolder("parent")
+        val lastModified = 12345L
+        val file = File(parent, "FileName")
+        file.createNewFile()
+        file.setLastModified(lastModified)
+
+        FileHelper.touchFile(parent, "FileName")
+
+        assertThat(file.lastModified()).isNotEqualTo(lastModified)
+    }
+
+    @Test
+    fun `renameOrMoveByCopying should move file`() {
+        val source = tempFolder.newFile("source")
+        val destination = tempFolder.newFile("destination")
+
+        FileHelper.renameOrMoveByCopying(source, destination)
+
+        assertThat(source.exists()).isFalse()
+        assertThat(destination.exists()).isTrue()
```

Comment:
This test would pass if the implementation of `FileHelper.renameOrMoveByCopying()` would simply delete the source file. The test passing doesn't give much confidence that `FileHelper.renameOrMoveByCopying()` actually does what it says it does.
The test could write something to the source file and then read the destination file after the move operation to see if the file contains the expected content. 

The common case for our app is that the destination file doesn't exist. So if we could write only one test, it should be one where the destination file doesn't exist. Ideally, of course, both cases would be tested.

Reply:
I added tests for both and ensure that the file content is verified.